src/containers/Search.jsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, useCallback } from 'react';
+import axios from 'axios';
+
+import Frame from '../components/Frame';
+import Header from '../components/Header';
+import Result from '../components/Result';
+import Loading from '../components/Loading';
+import ErrorMessage from '../components/ErrorMessage';
+
+const Search = () => {
+  const [fetchState, setFetchState] = useState('none');
+  const [value, setValue] = useState('');
+  const [user, setUser] = useState([]);
+  const [errorName, setErrorName] = useState('');
+
+  const fetchData = useCallback(async username => {
+    try {
+      const response = await axios(
+        `https://api.github.com/search/repositories?q=+user:${username}`,
+      );
+      setUser(response.data.items);
+      setFetchState('render');
+    } catch (error) {
+      setFetchState('error');
+    }
+  }, []);
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    setFetchState('loading');
+    setUser([]);
+    fetchData(value);
+    setErrorName(value);
```

Comment:
무조건 errorName을 설정하기보단, 이렇게 수정하면 어떨까요??
```suggestion
    try {
      const response = await axios(
        `https://api.github.com/search/repositories?q=+user:${username}`,
      );
      setUser(response.data.items);
      setFetchState('render');
    } catch (error) {
      setFetchState('error');
      setErrorName(value);
    }
  }, []);

  const handleSubmit = event => {
    event.preventDefault();
    setFetchState('loading');
    setUser([]);
    fetchData(value);
```

Reply:
오호 API 요청 보내는 쪽에서 에러메세지 설정하는게 맞는 것 같네요!